engine/src/main/java/io/zeebe/engine/processing/common/CatchEventBehavior.java

```diff
@@ -171,32 +178,29 @@ public void unsubscribeFromTimerEvent(
 
   private void subscribeToMessageEvent(
       final BpmnElementContext context,
-      final ExecutableCatchEvent handler,
-      final DirectBuffer extractedKey,
-      final DirectBuffer extractedMessageName,
+      final ExecutableCatchEvent catchEvent,
+      final DirectBuffer correlationKey,
+      final DirectBuffer messageName,
       final SideEffects sideEffects) {
 
     final long processInstanceKey = context.getProcessInstanceKey();
     final DirectBuffer bpmnProcessId = cloneBuffer(context.getBpmnProcessId());
     final long elementInstanceKey = context.getElementInstanceKey();
 
-    final DirectBuffer correlationKey = extractedKey;
-    final DirectBuffer messageName = extractedMessageName;
-    final boolean closeOnCorrelate = handler.shouldCloseMessageSubscriptionOnCorrelate();
     final int subscriptionPartitionId =
         SubscriptionUtil.getSubscriptionPartitionId(correlationKey, partitionsCount);
 
     subscription.setSubscriptionPartitionId(subscriptionPartitionId);
     subscription.setMessageName(messageName);
     subscription.setElementInstanceKey(elementInstanceKey);
-    subscription.setCommandSentTime(ActorClock.currentTimeMillis());
     subscription.setProcessInstanceKey(processInstanceKey);
     subscription.setBpmnProcessId(bpmnProcessId);
     subscription.setCorrelationKey(correlationKey);
-    subscription.setTargetElementId(handler.getId());
-    subscription.setCloseOnCorrelate(closeOnCorrelate);
-    // todo(zell): phil will migrate this via writing the open subscription
-    processInstanceSubscriptionState.put(subscription);
+    subscription.setElementId(catchEvent.getId());
+    subscription.setInterrupting(catchEvent.isInterrupting());
+
+    // TODO (saig0): the subscription should have a key (#2805)
+    stateWriter.appendFollowUpEvent(-1L, ProcessInstanceSubscriptionIntent.CREATING, subscription);
 
     sideEffects.add(
```

Comment:
More a question than a comment: Why is this executed as side effect? Why not write the command directly?

Reply:
Previously, we used the side effects to extract the behavior that should not be executed during reprocessing. In this case, the subscription may be created already. So, we don't want to send the command again.

After the migration, it may be okay to call it directly.